src/Elan.Friends/Services/FriendsService.cs

```diff
@@ -17,14 +19,40 @@ public FriendsService(IDataService dataService)
             _dataService = dataService;
         }
 
+        public async Task<FriendsRelation> CreateRelation(ElanUser firstUser, ElanUser secondUser)
+        {
+            var relation = new FriendsRelation
+            {
+                FirstUserId = firstUser.Id,
+                SecondUserId = secondUser.Id,
+                FirstUser = firstUser,
+                SecondUser = secondUser,
+                CreatedDate = DateTime.UtcNow
+            };
+
+            await _dataService.GetSet<FriendsRelation>().AddAsync(relation);
+            await _dataService.SaveDbAsync();
+            return relation;
+        }
+
         public async Task<List<ElanUser>> GetFriendsForUser(ElanUser user)
         {
-            // TODO: rework to fetch actual user friends
-            var result = await _dataService.GetSet<ElanUser>()
-                .Where(u => u.Id != user.Id)
+            var result = await _dataService.GetSet<FriendsRelation>()
+                .Include(u => u.FirstUser)
+                .Include(u => u.SecondUser)
+                .Where(u => u.FirstUser.Id == user.Id || u.SecondUser.Id == user.Id)
+                .Select(r => GetFriendUser(r, user))
                 .ToListAsync();
-
             return result;
         }
+
+        private ElanUser GetFriendUser(FriendsRelation r, ElanUser user)
```

Comment:
We should use more descriptive names than `r`.

Reply:
done